src/libs/blueprint/conduit_blueprint_mesh.cpp

```diff
@@ -4535,12 +4573,12 @@ mesh::matset::verify(const Node &matset,
             verify_number_field(protocol, matset, info, "volume_fractions"))
         {
             log::info(info, protocol, "detected uni-buffer matset");
+            // materials_map is not optional in this case, signal
+            // for opt check down the line
+            mat_map_is_optional = false;
 
             vfs_res &= verify_integer_field(protocol, matset, info, "material_ids");
-            // TODO(JRC): Add a more in-depth verifier for 'material_map' that
-            // verifies that it's one level deep and that each child child houses
-            // an integer-style array.
-            vfs_res &= verify_object_field(protocol, matset, info, "material_map");
+            vfs_res &= verify_matset_material_map(protocol,matset,info);
```

Comment:
I think I'd make this check separate, similar to what's done in the multi-buffer case; this way, the user can check for a valid `material_set` entry (i.e. an object with all integer values) without depending on a valid `volume_fractions` entry.

Reply:
I agree, I shifted the logic to outside